packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3212 @@
+      return true;
+    }());
+    _parent?._addChild(this);
+  }
+
+  @override
+  final _MenuHandleBase? _parent;
+
+  /// Returns true if this [MenuHandle]'s menu is open.
+  @override
+  bool get isOpen => _overlayEntry != null;
+
+  @override
+  Axis get _orientation => Axis.vertical;
+
+  final List<Widget> _menuChildren;
+  final FocusScopeNode _menuScopeNode;
+  final MenuStyle? _menuStyle;
+  final ButtonStyle? _buttonStyle;
+  final VoidCallback? _onOpen;
+  final VoidCallback? _onClose;
+  final FocusNode? _buttonFocusNode;
+  final Clip _menuClipBehavior;
+  final Offset _alignmentOffset;
+  final bool _ownsParent;
+  Offset? _globalMenuPosition;
+
+  @protected
+  OverlayEntry? _overlayEntry;
+
+  FocusNode? get _firstItemFocusNode {
+    assert(ChangeNotifier.debugAssertNotDisposed(this));
+    if (_menuScopeNode.context == null) {
+      return null;
+    }
+    final FocusTraversalPolicy policy =
+        FocusTraversalGroup.maybeOf(_menuScopeNode.context!) ?? ReadingOrderTraversalPolicy();
+    return policy.findFirstFocus(_menuScopeNode, ignoreCurrentFocus: true);
+  }
+
+  /// Open the menu, optionally at a global position.
+  ///
+  /// Call this when the menu should be shown to the user.
+  ///
+  /// The optional `position` argument will update the global coordinate where
+  /// the menu should be opened, taking into account the `alignmentOffset`, and
+  /// overriding any `globalMenuPosition` given to [createMaterialMenu].
+  ///
+  /// If `position` is not given, and no `globalMenuPosition` was given to
+  /// [createMaterialMenu], the menu will appear at the location that is
+  /// `alignmentOffset` away from the alignment origin specified by the
+  /// combination of the [MenuStyle.alignment] for the menu and the [MenuAnchor]
+  /// for the menu.
+  void open(BuildContext context, {Offset? position}) {
+    assert(ChangeNotifier.debugAssertNotDisposed(this));
+    if (isOpen && position == _globalMenuPosition) {
+      assert(_debugMenuInfo("Not opening $this because it's already open"));
+      return;
+    }
+    assert(_debugMenuInfo(
+        'Opening ${this}${_globalMenuPosition != null ? ' at ${position ?? _globalMenuPosition}' : ''}'));
+    final bool somethingWasOpen = _root.isOpen;
+    _parent?._closeChildren(); // Close all siblings.
+    assert(_overlayEntry == null);
+
+    _globalMenuPosition = position ?? _globalMenuPosition;
+    final _MenuAnchorMarker? anchor = _MenuAnchorMarker.maybeOf(context);
+    assert(
+        anchor != null || _globalMenuPosition != null,
+        'Unable to determine menu position.\n'
+        'Menus either need to have a globalMenuPosition set, or there needs to '
+        'be a MenuAnchor widget ancestor in the given context: $context');
+    assert(() {
+      if (anchor == null || _globalMenuPosition != null) {
+        return true;
+      }
+      final _MenuHandleMarker? marker = anchor.anchorKey.currentContext!.findAncestorWidgetOfExactType<_MenuHandleMarker>();
```

Comment:
Isn't _MenuHandleMarker an inheritedwidget that could/should be looked up via one of the `getFoo` methods instead of `findFoo`?